geth/mixins.py

```diff
@@ -50,8 +51,15 @@ def _get_file_logger(name: str, filename: str) -> logging.Logger:
     return logger
 
 
-class JoinableQueue(queue.Queue):
-    def __iter__(self):
+# only needed until we drop support for python 3.8
+if TYPE_CHECKING:
+    BaseQueue = queue.Queue[Any]
```

Comment:
I think you could `from __future__ import annotations` here to avoid this else if you want 🤷

Reply:
Doesn't work :(

Reply:
https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime

Reply:
Boo. Thanks for checking.